src/shell/commands/detect_hotkey.cpp

```diff
@@ -32,21 +33,21 @@ bool generate_hotkey_request(dsn::replication::detect_hotkey_request &req,
                              int partition_index,
                              std::string &err_info)
 {
-    if (!strcasecmp(hotkey_type.c_str(), "read")) {
+    if (dsn::utils::iequals(hotkey_type.c_str(), "read")) {
```

Comment:
how about define the parameter as `string_view`? so you it's not needed to convet to C string.

Reply:
Actually `string_view` is used not only for the string of the printable characters, but also binary bytes. Once `string_view` is used as the type of the parameters, we have to reimplement comparison ignoring cases based on `string_view` where binary bytes such as '\0' should be skipped. Therefore here we just declare other functions one parameter of which is declared as `std::string`.  Comparing two `std::string`-typed strings we can use `boost::iequals`.